internal/services/pubkey_service.go

```diff
@@ -24,7 +27,11 @@ func CreatePubkey(w http.ResponseWriter, r *http.Request) {
 
 	err = pkDao.Create(r.Context(), payload.Pubkey)
 	if err != nil {
-		renderError(w, r, payloads.NewDAOError(r.Context(), "create pubkey", err))
+		if db.IsUniqueConstraintError(err) != nil {
+			renderError(w, r, payloads.PubkeyAlreadyExistsError(r.Context(), err))
```

Comment:
Should we wrap it in DAO and return a nicer error? I mean this is fine, but I feel like letting pg errors bubble up feels bit wrong (eg, it's hard to stub them)

Reply:
Fact is, it already is a "nice error". We are taking advantage of error wrapping here, the `IsPostgresError` function actually drills down the wrapped errors until it finds the root cause (pg error). But what you actually have is `sqlx.NewCreateError` (our own type from dao/sqlx).